src/helm/proxy/clients/aleph_alpha_client.py

```diff
@@ -135,22 +153,20 @@ def do_it():
         )
 
     def decode(self, request: DecodeRequest) -> DecodeRequestResult:
-        """Make a request following https://docs.aleph-alpha.com/api/detokenize."""
-        raw_request = {
-            "model": request.tokenizer_name,
-            "token_ids": request.tokens,
-        }
-
+        """
+        Decode the tokens using Aleph Alpha's tokenizer library:
+        https://aleph-alpha-client.readthedocs.io/en/latest/aleph_alpha_client.html#aleph_alpha_client.Client.tokenizer
+        """
         try:
 
             def do_it():
-                result = self._send_request(AlephAlphaClient.DETOKENIZE_ENDPOINT, raw_request)
-                assert "result" in result, f"Invalid response: {result}"
-                return result
+                tokenizer: Tokenizer = self._get_tokenizer(request.tokenizer_name)
+                return {"result": tokenizer.decode(request.tokens)}
 
-            response, cached = self.cache.get(raw_request, wrap_request_time(do_it))
-        except (requests.exceptions.RequestException, AssertionError) as e:
-            error: str = f"AlephAlphaClient error: {e}"
+            cache_key = {"model": request.tokenizer_name, "token_ids": request.tokens}
```

Comment:
No action required: It would have been nice to indicate in the cache key what the type of the request is... but I suppose it would break compatibility now.